packages/next/src/builders/build/lib/create-package-json.ts

```diff
@@ -0,0 +1,76 @@
+import { BuilderContext } from '@angular-devkit/architect';
+import { createProjectGraph } from '@nrwl/workspace/src/core/project-graph';
+import { writeFileSync } from 'fs';
+import { join } from 'path';
+import { readJsonFile } from '@nrwl/workspace';
+import { NextBuildBuilderOptions } from '../../../utils/types';
+import { reactDomVersion, reactVersion } from '@nrwl/react/src/utils/versions';
+import { CSS_IN_JS_DEPENDENCIES } from '@nrwl/react';
+
+export async function createPackageJson(
+  options: NextBuildBuilderOptions,
+  context: BuilderContext
+) {
+  const rootPackageJson = readJsonFile(
+    join(context.workspaceRoot, 'package.json')
+  );
+
+  const allWorkspaceDeps = {
+    ...rootPackageJson.dependencies,
+    ...rootPackageJson.devDependencies
+  };
+
+  const outPackageJson = {
+    name: context.target.project,
+    version: '0.0.1',
+    scripts: {
+      start: 'next start'
+    },
+    dependencies: {
+      next: allWorkspaceDeps.next,
+      react: allWorkspaceDeps.react || reactVersion,
+      'react-dom': allWorkspaceDeps['react-dom'] || reactDomVersion
+    },
+    devDependencies: {}
+  };
+
+  const projGraph = createProjectGraph();
+
+  projGraph.dependencies[context.target.project].forEach(d => {
+    const p = projGraph.nodes[d.target];
+
+    if (p.type !== 'npm') return;
+
+    if (CSS_IN_JS_DEPENDENCIES[p.name]) {
+      // If this is styled-components or emotion then we may need to pull in
+      // additional dependencies
+      const { dependencies, devDependencies } = CSS_IN_JS_DEPENDENCIES[p.name];
+      outPackageJson.dependencies = {
+        ...outPackageJson.dependencies,
+        ...preserveVersonsFromWorkspace(dependencies, allWorkspaceDeps)
+      };
+      outPackageJson.devDependencies = {
+        ...outPackageJson.devDependencies,
+        ...preserveVersonsFromWorkspace(devDependencies, allWorkspaceDeps)
+      };
+    } else {
+      // Otherwise just add dependency to output package.json
+      outPackageJson.dependencies[p.name] = p.data.version;
+    }
+  });
+
+  writeFileSync(
+    join(options.outputPath, 'package.json'),
+    JSON.stringify(outPackageJson, null, 2)
+  );
+}
+
+function preserveVersonsFromWorkspace(nxDeps, allRootDeps) {
+  return Object.keys(nxDeps).reduce(
+    (acc, k) => {
+      acc[k] = allRootDeps[k] || acc[k];
+      return acc;
+    },
+    { ...nxDeps }
+  );
+}
```

Comment:
this is awesome - just out of curiosity, why do we need dynamic dependencies in package.json?

Would we not need just what's necessary to run the start command? Would we need anything else apart from next, react, and react-dom ?

Reply:
So I think to optimize for a monorepo, it'd be better to push only the dist folder. For example, `dist/apps/demo` instead of the root. We don't need to deploy the source code, but we will need to install dependencies needed by the compiled code.